Proofs/Popcount32.lean

```diff
@@ -79,18 +79,6 @@ theorem popcount32_sym_no_error (s0 s_final : ArmState)
   simp_all only [state_simp_rules, -h_run]
   -- Symbolic Simulation
   sym_n 27
```

Comment:
Correct me if I'm wrong, but `sym_n` still leaves `h_step_<i>` hyps. in the context, right? If so, are we ready to reduce goal clutter by clearing them away? After all, we have effects and non-effects hyps now.

Reply:
Awkward comment placement because the diff. reminded me of this...

Reply:
We could, except for the fact that this aggregation does not quite work cleanly with the VCG framework, where we simulate one step at a time. We could manually invoke the right (non-)effects hyps, but that is somewhat annoying to do by hand

Reply:
`sym_n` does generate `h_s{i}_<component>` hyps. that describe the `h_step_{i}` hyps in terms of reads, right? Those `h_s{i}_<component>` ought to work for manual aggregation needed in VCG proofs at the moment. 

The AbsVCGTandem proof would need to change to stop using `h_step_{i}` hyps. for aggregation and use `h_s{i}_<component>` hyps instead. I can try that after this PR's merged, and if I find myself not needing the `h_step_{i}` hyps at all, then let's clear them out.